app/javascripts/app.js

```diff
@@ -108,6 +117,10 @@ async function runAt(address) {
 
           for (let i = numCandidates - 1; i >= 0; i--) {
             let candidateName = web3c.utils.toUtf8(response[i]).toString();
+            if (!nameRegex.test(candidateName)) {
```

Comment:
This is great.  Somebody who loads this page can edit the javascript to disable this check via Chrome's dev tools before interacting with the page, and still inject in bad data into the contract.  This check prevents it from causing problems.

A comment along these lines would be nice, as both an educational opportunity and as defense against a future optimizer who sees the check at the contract deploy and thinks to optimize things by removing this apparently redundant check.